matching-service/src/main/java/com/local_dating/matching_service/application/MatchingScheduleService.java

```diff
@@ -63,5 +71,74 @@ public void updateSchedule(final Long id, final Long userId, final MatchingSched
             return el;
         }).orElseThrow(() -> new BusinessException(MessageCode.MATCHING_SCHEDULE_NOT_FOUND));
+
+        checkMatchingSchedule(id);
+    }
+
+    @Transactional(propagation = Propagation.NESTED)
+    public void checkMatchingSchedule(final Long id) {
+
+        List<MatchingSchedule> scheduleVO1;
+        List<MatchingSchedule> scheduleVO2;
+
+        List<LocalDate> matchingDateList1 = new ArrayList<>();
+        List<LocalDate> matchingDateList2 = new ArrayList<>();
+
+        Optional<MatchingVO> matchingVO = matchingService.getMatchingInfo(id);
+
+        scheduleVO1 = matchingScheduleRepository.findMatchingSchedulesByMatchingIdAndUserId(id, matchingVO.get().requId()); //매칭요청자가 보낸 스케줄
+        scheduleVO2 = matchingScheduleRepository.findMatchingSchedulesByMatchingIdAndUserId(id, matchingVO.get().recvId()); //매칭수락자가 보낸 스케줄
+        /*scheduleVO1 = matchingScheduleRepository.findMatchingSchedulesByIdAndUserId(id, matchingVO.get().requId());
+        scheduleVO2 = matchingScheduleRepository.findMatchingSchedulesByIdAndUserId(id, matchingVO.get().requId());*/
+
+        scheduleVO1.stream().forEach(el -> matchingDateList1.add(el.getMatchingDate()));
+        scheduleVO2.stream().forEach(el -> matchingDateList2.add(el.getMatchingDate()));
+
+        matchingDateList1.retainAll(matchingDateList2);
+        //scheduleVO1.retainAll(scheduleVO2);
+
+        if (!matchingDateList1.isEmpty()) {
+            scheduleVO1.stream().filter(el -> matchingDateList1.contains(el.getMatchingDate()))
+                    .forEach(el -> el.setScheduleStatus(MatchingType.SAME.getCode()));
+            scheduleVO2.stream().filter(el -> matchingDateList2.contains(el.getMatchingDate()))
+                    .forEach(el -> el.setScheduleStatus(MatchingType.SAME.getCode()));
+        }
+
+        /*
+        matchingService.getMatchingInfo(id).map(el->{
+            scheduleVO1 = matchingScheduleRepository.findMatchingScheduleByIdAndUserId(id, el.requId());
+            scheduleVO2 = matchingScheduleRepository.findMatchingScheduleByIdAndUserId(id, el.recvId());
+            return el;
+        });
+        */
+    }
+
+    @Transactional
+    public void requestMatchingSchedule(final Long id, final Long scheduleId) {
+        matchingScheduleRepository.findMatchingScheduleByIdAndScheduleIdAndScheduleStatus(id, scheduleId, MatchingType.SAME.getCode())
+                //matchingScheduleRepository.findMatchingScheduleByIdAndScheduleId(id, scheduleId)
+                .map(el -> {
+                    el.setScheduleStatus(MatchingType.REQUEST.getCode());
+                    return el;
+                }).orElseThrow(() -> new BusinessException(MessageCode.MATCHING_SCHEDULE_NOT_FOUND));
+    }
+
+    @Transactional
+    public void acceptMatchingSchedule(final Long id, final Long scheduleId) {
+        matchingScheduleRepository.findMatchingScheduleByIdAndScheduleIdAndScheduleStatus(id, scheduleId, MatchingType.REQUEST.getCode())
```

Comment:
this.scheduling(final Long id, final Long scheduleId, MatchingType.REQUEST, MatchingType.MATCHED)